# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: cee7eaa8c402
# Full Hash: cee7eaa8c402e375e4c03fa3b65404915c5f33eb
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-11-03 21:39:33
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 4: Untie frame iteration from Frame::tls. r=lth
#   
#   Here we replace usage of Frame::tls in frame iteration with GetNearestEffectiveTls.
#   We also maintain current tls for frame iteration object to not to call GetNearestEffectiveTls everytime.
#   
# ==============================================================================

diff -r aeacd48f565e -r cee7eaa8c402 js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Tue Nov 03 10:22:41 2020 +0000
+++ b/js/src/jit/MacroAssembler.h	Tue Nov 03 10:25:14 2020 +0000
@@ -4416,10 +4416,13 @@
       : ABIArgIterBase<VecT, WasmABIArgGenerator>(types) {}
 };
 
-wasm::TlsData* ExtractCalleeTlsFromFrameWithTls(wasm::Frame* fp);
-wasm::TlsData* ExtractCallerTlsFromFrameWithTls(wasm::Frame* fp);
-
 }  // namespace jit
+
+namespace wasm {
+TlsData* ExtractCalleeTlsFromFrameWithTls(Frame* fp);
+TlsData* ExtractCallerTlsFromFrameWithTls(Frame* fp);
+}  // namespace wasm
+
 }  // namespace js
 
 #endif /* jit_MacroAssembler_h */